Clean up metamist readme and installation instructions (#1105)
populationgenomics/metamist: docs/installation.md

**Note** this is a Metamist Developer guide, not a Metamist user guide. These steps are only required if you need to test a script on a local version of metamist or work on contributions to the metamist code base. If you just need to access and work with metamist data see [the instructions here](README.md)

## Clone the codebase

```bash
# Clone the repo
git clone git@github.com:populationgenomics/metamist.git

# enter the metamist repo directory
cd metamist
```

## API setup

### Install python requirements

It is recommended to use [pyenv](https://github.com/pyenv/pyenv) and [virtualenv](https://virtualenv.pypa.io/en/latest/installation.html) to manage the python version and virtual environment for metamist. Metamist currently is deployed with python 3.11 so it is best to use that locally to ensure your changes are compatible.

```bash
# install python 3.11
pyenv install 3.11

# select python 3.11 for your current shell
pyenv shell 3.11

# set up virtualenv with python 3.11
virtualenv .venv -p 3.11

# activate virtualenv
source .venv/bin/activate

# install requirements
pip install --no-deps -r requirements-dev.txt
```

### Database setup

#### Running MariaDB

> Metamist uses a MariaDB 11 database. Docker is probably the easiest way to run the metamist MariaDB database locally.

Reviewer: ```suggestion
Metamist uses a MariaDB 11 database. Docker is the easiest way to run the Metamist MariaDB database locally.
```
Nit, instructional rather than narrative.

We have found that [OrbStack](https://orbstack.dev/) is faster and easier to use than [Docker Desktop](https://docs.docker.com/desktop/) but either should work fine.

> It is also possible to install MariaDB stanalone using [homebrew](brew.sh) or another package manager.

Reviewer: ```suggestion
You can also install MariaDB standalone using [homebrew](brew.sh) or another package manager.
```
Nit, typo, tighten.

To start the database, run:

```bash
docker run --name mariadb-p3306 -e MYSQL_ALLOW_EMPTY_PASSWORD=1 -p 3306:3306 -d docker.io/library/mariadb:11.7.2
```

> If you need to run on a different port as `3306` is already taken you can specify that in the port mapping.

Reviewer: ```suggestion
If port 3306 is already in use, you can specify a different port in the mapping. For example:
```
nit, tighten
